lib/fast_excel.rb

```diff
@@ -410,39 +483,55 @@ def append_row(values, formats = nil)
     end
 
     def last_row_number
-      defined?(@last_row_number) ? @last_row_number : -1
+      @last_row_number
     end
 
     def increment_last_row_number!
-      @last_row_number = last_row_number + 1
+      @last_row_number += 1
     end
 
     def set_column(start_col, end_col, width, format = nil)
       super(start_col, end_col, width, format)
+
+      start_col.upto(end_col) do |i|
+        @col_formats[i] = format
+      end if format
     end
 
     def set_column_width(col, width)
-      set_column(col, col, width, nil)
+      set_column(col, col, width, @col_formats[col])
     end
 
     def set_columns_width(start_col, end_col, width)
-      set_column(start_col, end_col, width, nil)
+      #set_column(start_col, end_col, width, nil)
+      start_col.upto(end_col) do |i|
+        set_column_width(i, width)
+      end
+    end
+
+#    def get_col_format(cell_number)
+#      if self[:col_formats_max] >= cell_number
+#        format_pointer = self[:col_formats] + cell_number * Libxlsxwriter::Format.size
+#        format_pointer.null? ? nil : Libxlsxwriter::Format.new()
+#      else
+#        nil
+#      end
+#    end
+
+    def close
+      if auto_width?
+        @column_widths.each do |num, width|
+          #p [:set_column_width, num, width]
+          set_column_width(num, width + 0.2)
```

Comment:
I tried Arial and Calibri at different point sizes, and found that 0.2 still cuts off some strings, particularly noticeable on a column of just dates.  After changing to use 1 instead, it gives it enough margin.
set_column_width(num, width + 1)

Reply:
I'm finding that column widths are always too small even with width + 1 set here.